From f120ddabf4e46661c081dbbaffa1419c221df7cd Mon Sep 17 00:00:00 2001
From: Simon Turcotte-Langevin <sturcottelangevin@bhvr.com>
Date: Sun, 20 Aug 2023 19:41:04 -0400
Subject: [PATCH] Fixup top level directory flag

---
 CMakeLists.txt | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 80d0aca..61d84ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,31 +5,25 @@
 cmake_minimum_required(VERSION 3.11)
 project(cppast VERSION 0.1.0)
 
-if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
-    set(is_top_level_project TRUE)
-else()
-    set(is_top_level_project FALSE)
-endif()
-
 # options
 option(CPPAST_BUILD_TEST "whether or not to build the tests" OFF)
 option(CPPAST_BUILD_EXAMPLE "whether or not to build the examples" OFF)
 option(CPPAST_BUILD_TOOL "whether or not to build the tool" OFF)
 
-if(${CPPAST_BUILD_TEST} OR ${is_top_level_project})
+if(${CPPAST_BUILD_TEST})
     set(build_test ON)
     set(CMAKE_EXPORT_COMPILE_COMMANDS ON) # for the self integration test
 else()
     set(build_test OFF)
 endif()
 
-if(${CPPAST_BUILD_EXAMPLE} OR ${is_top_level_project})
+if(${CPPAST_BUILD_EXAMPLE})
     set(build_example ON)
 else()
     set(build_example OFF)
 endif()
 
-if(${CPPAST_BUILD_TOOL} OR ${is_top_level_project})
+if(${CPPAST_BUILD_TOOL})
     set(build_tool ON)
 else()
     set(build_tool OFF)
-- 
2.35.1.windows.2

