From 976ff16b2c6bace194eb9533a7c38cd290b00613 Mon Sep 17 00:00:00 2001
From: Simon Turcotte-Langevin <sturcottelangevin@bhvr.com>
Date: Sun, 20 Aug 2023 20:09:29 -0400
Subject: [PATCH] Fixup tiny-process-library

---
 external/external.cmake | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/external/external.cmake b/external/external.cmake
index 556d847..195b3e2 100644
--- a/external/external.cmake
+++ b/external/external.cmake
@@ -19,7 +19,10 @@ endif()
 find_package(Threads REQUIRED QUIET)

 # create a target here instead of using the one provided
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/tpl)
+find_package(cppast_tiny_process CONFIG)
+if (NOT ${cppast_tiny_process_FOUND})
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/tpl)
+endif ()
 add_library(_cppast_tiny_process INTERFACE)
 target_include_directories(_cppast_tiny_process INTERFACE ${tiny_process_dir})
 target_link_libraries(_cppast_tiny_process INTERFACE tiny-process-library::tiny-process-library Threads::Threads)
-- 
2.35.1.windows.2

